### DISCUSSION TITLE
Support for Gutenberg Blocks

### DISCUSSION THREAD

**Discussion by Bowriverstudio** [wpengine/atlas-content-modeler#543] [Ideas]
Hello,

Similar to what is requested on the discord chat https://discord.com/channels/836253505944813629/970765953979809822/970847222730203206

I'm making a Real estate site.  I'd like to use ACM - but the description field needs to support Gutenberg blocks.  Any suggestions?

Thanks,
Maurice

**Comment by mindctrl** [comment]
@Bowriverstudio thanks for the suggestion. Over the next 2 weeks, we're going to be building a proof of concept for using blocks in ACM. I don't know yet when we might launch that, but we're working on it. One idea we have is to make it a separate field type. Another idea is to make it an option on an existing field type (like the Rich Text field). I'm curious if you have any thoughts or preferences?

**Reply by Bowriverstudio** [reply]
@mindctrl  Thank you for the update.  

Both suggestions would work for me.  It might be easier to add a new field type, as modifying the Rich Text Field may cause backward compatibility issues.

In my case, I have two sections that I'd like to add blocks to and ideally lock them to a template (but this is not required).

Something like this

```php
/**
 * Plugin Name: Gutenberg Blocks
 * Author: Muhammad Owais Alam
 * Description: This plugin allows users to add a WordPress Gutenberg custom template for the Gutenberg editor
 * Version: 1.0.0
 */

add_action( 'init', function() {
    $args = array(
        'public' => true,
        'label'  => 'News',
        'show_in_rest' => true,
        'template_lock' => 'all',
        'template' => array(
            array( 'core/paragraph', array(
                'placeholder' => 'Breaking News',
            ) ),
            array( 'core/image', array(
                'align' => 'right',
            ) ),
        ),


    );

    register_post_type( 'news', $args );
} );
```


CHeers I and I would be happy to help test the POC.


Best Regards,
Mauriice

**Reply by Bowriverstudio** [reply]
@mindctrl Hello any updates on this?
